packages/ts-docs-gen/src/api-items/api-definition-base.ts

```diff
@@ -0,0 +1,77 @@
+import { Contracts, ExtractDto } from "ts-extractor";
+
+import { BaseApiItemClass } from "../abstractions/base-api-item";
+import { GeneratorHelpers } from "../generator-helpers";
+import { SerializedApiDefinition } from "../contracts/serialized-api-item";
+import { Helpers } from "../utils/helpers";
+import { ApiItemReference } from "../contracts/api-item-reference";
+import { ApiDefinitions } from "./api-definition-list";
+import { ApiTypes } from "./api-type-list";
+
+/**
+ * Base definition class with helper functions.
+ */
+export abstract class ApiDefinitionBase<TKind extends Contracts.ApiBaseItemDto = Contracts.ApiBaseItemDto> extends BaseApiItemClass<TKind>
+    implements SerializedApiDefinition<TKind> {
+
+    constructor(extractedData: ExtractDto, apiItem: TKind, private reference: ApiItemReference) {
+        super(extractedData, apiItem);
+    }
+
+    public get Reference(): ApiItemReference {
+        return this.reference;
+    }
+
+    private parentItem: ApiDefinitions | undefined;
+
+    public get ParentItem(): ApiDefinitions | undefined {
+        if (this.Data.ParentId != null) {

```

Comment:
Make `ParentItem` lazier. Second call should not call `this.GetSerializedApiDefinition(parentReference);`